### PR TITLE
Add CI icons to the readme

### PR DESCRIPTION
**Pull request by grynspan**
Add CI icons to the readme (using PR jobs for now.)

### Motivation:

CI icons provide at-a-glance status information about supported platforms.

### Modifications:

Modifies the README file to include Jenkins CI icons for macOS and Linux jobs.

### Result:

Build icons will appear in the supported platforms table for macOS and Linux.

Looks like:

Screenshot 2023-09-22 at 10 25 42 AM

Resolves rdar://115073005.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci please test

File: README.md [hunk 1]
```diff
@@ -47,11 +47,11 @@ for various platforms:
 
 | **Platform** | **Status** |
 |---|---|
-| **macOS** | Supported |
+| **macOS** | [![Build Status](https://ci.swift.org/buildStatus/icon?job=pr-swift-testing-macos)](https://ci.swift.org/job/pr-swift-testing-macos/) |
 | **iOS** | Supported |
 | **watchOS** | Supported |
 | **tvOS** | Supported |
-| **Ubuntu 22.04** | Supported |
+| **Ubuntu 22.04** | [![Build Status](https://ci.swift.org/buildStatus/icon?job=pr-swift-testing-linux-main)](https://ci.swift.org/job/pr-swift-testing-linux-main/) |
```
Comment (stmontgomery) [inline README.md:54]: I wonder if we should break out these badges into a third column in the table, so we can still show their level of support status?

Also, the title of this section is currently "Supported platforms" but I wonder if now we ought to expand that to a broader title. Some other projects display their CI status more prominently towards the top of the README, and I thought once we began including CI we might do that as well.

Comment (grynspan) [reply README.md:54]: I can put it in a third "Build Status" column maybe?

I don't have any preferences about the title of the section or its position.

Comment (stmontgomery) [reply README.md:54]: Sure - how about these for the column titles?
```
| **Platform** | **CI Status** | **Support Status** |
```

Comment (grynspan) [reply README.md:54]: I buy it.

**Comment by grynspan** [comment]
@swift-ci please test

**Review by stmontgomery — APPROVED** [review]

**Comment by grynspan** [comment]
We've asked our colleague to set up additional Jenkins jobs for the main branch and will hold this PR until they've been enabled.

**Comment by grynspan** [comment]
@swift-ci please test

**Pull request merged**